src/kernel/thread.c

```diff
@@ -346,6 +349,11 @@ void schedule(void)
     awaken();
 #endif
 
+#ifdef CONFIG_KERNEL_MCS
+    bool_t switchedDomain = false;
+#else
+    bool_t switchedDomain __attribute__((unused));
+#endif
```

Comment:
There's an `UNUSED` macro that would be preferable to spelling out `__attribute__((unused))`.

Also, I think `UNUSED` is harmless when the variable is actually used. If so, this whole `#ifdef` could be replaced with `UNUSED bool_t switchedDomain = false;`. For example, in [cteDeleteOne](https://github.com/seL4/seL4/blob/7e798a15ba4c1a8beb94be1b6916c5869a4e9916/src/object/cnode.c#L731-L740), `fc_ret` is only "used" in debug builds. Even though it's assigned in every build, the assigned value is only read in the `assert`, which only appears in debug builds.